monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClient.java

```diff
@@ -31,12 +42,28 @@ public SingleIpHttpClient(HttpClient httpClient, InetAddress inetAddress, Server
         } catch (URISyntaxException | MalformedURLException e) {
             throw new IllegalStateException("Cannot build health URI from " + serverConfiguration, e);
         }
+
+        this.healthy = new AtomicBoolean();
+
+        final long connectionHealthCheckPeriodInSeconds = serverConfiguration.getConnectionHealthCheckPeriodInSeconds();
+
+        this.scheduledFuture = scheduledExecutorService.scheduleAtFixedRate(
```

Comment:
it could be nice to be able to disable the automatic refresh by settings a specific value like -1 or using a specific flag ?

Reply:
Okay

Reply:
What about doing this change in a separate MR? :)

Reply:
Just because it requires more tests and changes in the logic, it would be more handy to have it done as a second step

Reply:
ok fine for me